client/task-list/tasks/PaymentGatewaySuggestions/index.js

```diff
@@ -22,65 +22,76 @@ import './plugins/Bacs';
-						: {};
-
-					const enrichedSuggestion = {
-						installed: !! installedPaymentGateways[ id ],
-						postInstallScripts:
-							installedGateway.post_install_scripts,
-						enabled: installedGateway.enabled || false,
-						needsSetup: installedGateway.needs_setup,
-						settingsUrl: installedGateway.settings_url,
-						connectionUrl: installedGateway.connection_url,
-						setupHelpText: installedGateway.setup_help_text,
-						title: installedGateway.title,
-						requiredSettings: installedGateway.required_settings_keys
-							? installedGateway.required_settings_keys
-									.map(
-										( settingKey ) =>
-											installedGateway.settings[
-												settingKey
-											]
-									)
-									.filter( Boolean )
-							: [],
-						...suggestion,
-					};
-
-					map.set( id, enrichedSuggestion );
-					return map;
-				}, new Map() );
-
-			return {
-				getPaymentGateway: select( PAYMENT_GATEWAYS_STORE_NAME )
-					.getPaymentGateway,
-				getOption: select( OPTIONS_STORE_NAME ).getOption,
-				isResolving: select( ONBOARDING_STORE_NAME ).isResolving(
-					'getPaymentGatewaySuggestions'
+	const {
+		getPaymentGateway,
+		fetchedPaymentGateways,
+		isResolving,
+	} = useSelect( ( select ) => {
+		const installedPaymentGateways = select( PAYMENT_GATEWAYS_STORE_NAME )
+			.getPaymentGateways()
+			.reduce( ( map, gateway ) => {
+				map[ gateway.id ] = gateway;
+				return map;
+			}, {} );
+
+		const mappedSuggestions = select( ONBOARDING_STORE_NAME )
+			.getPaymentGatewaySuggestions()
+			.reduce( ( map, suggestion ) => {
+				const { id } = suggestion;
+				const installedGateway = installedPaymentGateways[
+					suggestion.id
+				]
+					? installedPaymentGateways[ id ]
+					: {};
+
+				const enrichedSuggestion = {
+					installed: !! installedPaymentGateways[ id ],
+					postInstallScripts: installedGateway.post_install_scripts,
+					enabled: installedGateway.enabled || false,
+					needsSetup: installedGateway.needs_setup,
+					settingsUrl: installedGateway.settings_url,
+					connectionUrl: installedGateway.connection_url,
+					setupHelpText: installedGateway.setup_help_text,
+					title: installedGateway.title,
+					requiredSettings: installedGateway.required_settings_keys
+						? installedGateway.required_settings_keys
+								.map(
+									( settingKey ) =>
+										installedGateway.settings[ settingKey ]
+								)
+								.filter( Boolean )
+						: [],
+					...suggestion,
+				};
+
+				map.set( id, enrichedSuggestion );
+				return map;
+			}, new Map() );
+
+		return {
+			getPaymentGateway: select( PAYMENT_GATEWAYS_STORE_NAME )
+				.getPaymentGateway,
+			getOption: select( OPTIONS_STORE_NAME ).getOption,
+			isResolving: select( ONBOARDING_STORE_NAME ).isResolving(
+				'getPaymentGatewaySuggestions'
+			),
+			fetchedPaymentGateways: mappedSuggestions,
+		};
+	}, [] );
+
+	const paymentGateways = useMemo( () => fetchedPaymentGateways, [
+		fetchedPaymentGateways.size,
```

Comment:
One of the issues with using `.size()` is that we won't get updates to the current list of enriched gateways.  We could also seem some edge cases in the future where the suggestions change, but are the same number of items and don't appear due to this memoization.